bin/run-amp-validator.js

```diff
@@ -0,0 +1,228 @@
+                                }
+                                i++;
+                                resolve();
+                            }).catch(function (reason) {
+                                i++;
+                                ourErrors.push(reason);
+                                console.error('Error or timeout', reason);
+                                reject(reason);
+                            });
+                    }
+                });
+        });
+    });
+};
+
+describe('AMP Validation Suite', function() {
+    this.timeout(20000);
+
+    before( function() {
+        var child = exec('wp post list --post_type=post --posts_per_page=-1 --post_status=publish --post_password="" --format=json --fields=url --quiet --skip-plugins=wordpress-importer');
+
+        child.stdout.on('data', function (data) {
+            var items = JSON.parse(data.trim());
+
+            for (var i=0 , len = items.length; i < len; i++ ) {
+                var item = items[i];
+                if ( '/' != item['url'].slice(-1) ) {
+                    item['url'] = item['url']+"/";
+                }
+                testUrls.push( item['url']+"amp/" );
+
+            }
+
+            /**
+             * Control URLs for Testing purposes
+             * comment out for production usage
+             */
+            // var localBaseURL = url.parse(testUrls[0]);
+            // localBaseURL = localBaseURL.protocol + "//" + localBaseURL.hostname;
+            // testUrls.push(localBaseURL + '/wp-content/plugins/amp-wp/tests/assets/success.html');
+            // testUrls.push(localBaseURL + '/wp-content/plugins/amp-wp/tests/assets/404.html');
+            // testUrls.push(localBaseURL + '/wp-content/plugins/amp-wp/tests/assets/failure.html');
+
+        });
+        child.stderr.on('data', function (data) {
+
+        });
+
+        return promiseFromChildProcess(child)
+        .then(function () {
+                console.log("Hang tight, we are going to test "+testUrls.length+" urls...");
+            }, function (err) {
+                console.log('Child Exec rejected: ' + err);
+        }).then(function() {
+
+            return validateUrls( testUrls );
+
+        }).then(function(){
+
+            if (ourErrors.length > 0) {
+
+                //call our output errors function to print the errors in the console.
+                outputErrors( ourErrors );
+
+            }
+
+        });
+    });
+
+    it('Get URLs from WP', function(){
+        testUrls.length.should.not.equal(0);
+    });
+
+    it('Get All Validation Results', function(){
+        ourResults.length.should.equal(testUrls.length);
+    });
+
+    it('All URLs correctly validate', function(){
+        ourResults.length.should.equal(testUrls.length);
+        ourResults.should.all.be.equal('PASS');
+    });
+    it('No Errors found', function(){
+        ourResults.length.should.equal(testUrls.length);
+        ourErrors.length.should.equal(0);
+    });
```

Comment:
It feels a bit odd to run assertions for the test up bits; might be cleaner just to throw an error if getting the URLs fails, for example.

Also, could we not run each URL validation as a separate assertion?